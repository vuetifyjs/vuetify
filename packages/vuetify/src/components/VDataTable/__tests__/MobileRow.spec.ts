// @ts-nocheck
/* eslint-disable */

// import MobileRow from '../MobileRow'
import {
  mount,
  Wrapper,
  MountOptions,
} from '@vue/test-utils'
// import Vue from 'vue'

describe.skip('MobileRow', () => {
  type Instance = InstanceType<typeof MobileRow>
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>
  beforeEach(() => {
    mountFunction = (options?: MountOptions<Instance>) => {
      return mount(MobileRow, options)
    }
  })

  it('should render without slots', () => {
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { text: 'Petrol', value: 'petrol' },
            { text: 'Diesel', value: 'diesel' },
          ],
          item: {
            petrol: 0.68,
            diesel: 0.65,
          },
        },
      },
    })

    expect(wrapper.findAll('tr')).toHaveLength(1)
    expect(wrapper.findAll('td')).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render non-string values', () => {
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { value: 'string' },
            { value: 'number' },
            { value: 'array' },
            { value: 'boolean' },
            { value: 'object' },
            { value: 'undefined' },
            { value: 'null' },
          ],
          item: {
            string: 'string',
            number: 12.34,
            array: [1, 2],
            boolean: false,
            object: { foo: 'bar' },
            null: null,
          },
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with regular slots', () => {
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { text: 'Petrol', value: 'petrol' },
            { text: 'Diesel', value: 'diesel' },
          ],
        },
      },
      slots: {
        petrol: '<p class="test">$0.68</p>',
        diesel: '<p class="test">$0.65</p>',
      },
    })

    expect(wrapper.findAll('tr')).toHaveLength(1)
    expect(wrapper.findAll('td')).toHaveLength(2)
    expect(wrapper.findAll('p.test')).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render with scoped slots', () => {
    const vm = new Vue()
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { text: 'Petrol', value: 'petrol' },
            { text: 'Diesel', value: 'diesel' },
          ],
          item: {
            petrol: 0.68,
            diesel: 0.65,
          },
        },
      },
      scopedSlots: {
        petrol: props => vm.$createElement('p', { staticClass: `test ${props.header.value}` }, [props.value]),
        diesel: props => vm.$createElement('p', { staticClass: `test ${props.header.value}` }, [props.value]),
      },
    })

    expect(wrapper.findAll('tr')).toHaveLength(1)
    expect(wrapper.findAll('td')).toHaveLength(2)
    expect(wrapper.findAll('p.test')).toHaveLength(2)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render without header when hideDefaultHeader: true', () => {
    const wrapper = mountFunction({
      context: {
        props: {
          headers: [
            { text: 'Petrol', value: 'petrol' },
            { text: 'Diesel', value: 'diesel' },
          ],
          hideDefaultHeader: true,
          item: {
            petrol: 0.68,
            diesel: 0.65,
          },
        },
      },
    })

    expect(wrapper.findAll('tr')).toHaveLength(1)
    expect(wrapper.findAll('td')).toHaveLength(2)
    expect(wrapper.findAll('.v-data-table__mobile-row__header')).toHaveLength(0)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
