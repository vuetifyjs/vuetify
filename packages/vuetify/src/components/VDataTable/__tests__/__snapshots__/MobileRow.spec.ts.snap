// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileRow should render non-string values 1`] = `
<tr class="v-data-table__mobile-table-row">
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
    </div>
    <div class="v-data-table__mobile-row__cell">
      string
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
    </div>
    <div class="v-data-table__mobile-row__cell">
      12.34
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
    </div>
    <div class="v-data-table__mobile-row__cell">
      1,2
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
    </div>
    <div class="v-data-table__mobile-row__cell">
      false
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
    </div>
    <div class="v-data-table__mobile-row__cell">
      [object Object]
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
    </div>
    <div class="v-data-table__mobile-row__cell">
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
    </div>
    <div class="v-data-table__mobile-row__cell">
    </div>
  </td>
</tr>
`;

exports[`MobileRow should render with regular slots 1`] = `
<tr class="v-data-table__mobile-table-row">
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
      Petrol
    </div>
    <div class="v-data-table__mobile-row__cell">
      <p class="test">
        $0.68
      </p>
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
      Diesel
    </div>
    <div class="v-data-table__mobile-row__cell">
      <p class="test">
        $0.65
      </p>
    </div>
  </td>
</tr>
`;

exports[`MobileRow should render with scoped slots 1`] = `
<tr class="v-data-table__mobile-table-row">
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
      Petrol
    </div>
    <div class="v-data-table__mobile-row__cell">
      <p class="test petrol">
        0.68
      </p>
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
      Diesel
    </div>
    <div class="v-data-table__mobile-row__cell">
      <p class="test diesel">
        0.65
      </p>
    </div>
  </td>
</tr>
`;

exports[`MobileRow should render without header when hideDefaultHeader: true 1`] = `
<tr class="v-data-table__mobile-table-row">
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__cell">
      0.68
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__cell">
      0.65
    </div>
  </td>
</tr>
`;

exports[`MobileRow should render without slots 1`] = `
<tr class="v-data-table__mobile-table-row">
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
      Petrol
    </div>
    <div class="v-data-table__mobile-row__cell">
      0.68
    </div>
  </td>
  <td class="v-data-table__mobile-row">
    <div class="v-data-table__mobile-row__header">
      Diesel
    </div>
    <div class="v-data-table__mobile-row__cell">
      0.65
    </div>
  </td>
</tr>
`;
